src/tests/integration/monitor-tabs.test.ts

```diff
@@ -0,0 +1,200 @@
+import { expect } from 'chai'
+import { createMocks } from './mocks'
+import { runBackground } from './steps/run-background'
+
+describe.only('monitor tabs', () => {
+  const mocks = createMocks()
+
+  runBackground(mocks, { alreadyAuthenticated: true })
+
+  describe('chrome.tabs.onCreated', () => {
+    it("adds the newly created tab to the store's state", () => {
+      expect(mocks.chrome.tabs.onCreated.addListener).to.have.callCount(1)
+      const [callback] = mocks.chrome.tabs.onCreated.addListener.firstCall.args
+      callback({
+        id: 18,
+        windowId: 1,
+        active: false,
+        url: 'https://new.com/abc',
+      })
+      const state = mocks.getState()
+      const tab = state.tabs.get(18)!
+      expect(tab.id).to.equal(18)
+      expect(tab.windowId).to.equal(1)
+      expect(tab.domain).to.equal('new.com')
+      expect(tab.url).to.equal('https://new.com/abc')
+    })
+  })
+
+  describe('chrome.tabs.onRemoved', () => {
+    it("removes a tab from the store's state", () => {
+      const [callbackOnRemove] = mocks.chrome.tabs.onRemoved.addListener.firstCall.args
+
+      const tabId = 11
+      const removeInfo = {
+        id: 11,
```

Comment:
Generally looks good, although let's reduce the amount of code we need by defining only the variables/properties that are essential to the test. When we look we see `removeInfo` is never even used, so we can simplify like so.

```typescript
const toRemoveId = 11
expect(mocks.getState().tabs.get(toRemoveId)).to.not.equal(undefined)
const [callback] = mocks.chrome.tabs.onRemoved.addListener.firstCall.args
callback(toRemoveId)
expect(mocks.getState().tabs.get(toRemoveId)).to.equal(undefined)
```

Reply:
But the listener callback function asking for tabId and removeInfo, right? 
chrome.tabs.onRemoved.addListener((tabId, removeInfo) => dispatch['chrome.tabs.onRemoved']({ tabId, removeInfo }))

Reply:
Refactor as instructed. Test passed